crates/red_knot_python_semantic/src/semantic_index/builder.rs

```diff
@@ -738,9 +752,55 @@ where
                 is_star,
                 range: _,
             }) => {
+                // Save the state prior to visiting any of the `try` block.
+                //
+                // Potentially none of the `try` block could have been executed prior to executing
+                // the `except` block(s), so we will merge this state with all of the intermediate
+                // states during the `try` block before visiting the `except` blocks.
+                let pre_try_block_state = self.flow_snapshot();
+                self.try_block_contexts().push_try_block();
```

Comment:
What's the reason that we need to "emulate" a stack here instead of making use of the function's stack to allocate the try blocks?

I looked at `TryBlockContextsStack` and I didn't spot anything that obviously requires a `Vec` (e.g. because we would need to inspect the entire stack.)

Reply:
I didn't understand this suggestion (or the similar one in the top-level comment) so I checked in with Micha off-line; it sounds like this suggestion was based on a misunderstanding of what this PR does (and needs to do), in terms of tracking stack of try-blocks per scope (i.e. it currently uses a nested vec of vecs). This is effectively suggesting to go back to the previous `std::mem::replace` approach, which isn't sufficient because it doesn't clear the try blocks on scope-push and restore them on scope-pop.